File: FlowerAppAPI/FlowerAppAPI/Models/Product.cs
using System;
using System.Collections.Generic;

namespace FlowerAppAPI.Models;

public partial class Product
{
    public long Idproduct { get; set; }

    public string? Name { get; set; }

    public string? Description { get; set; }

    public decimal? Price { get; set; }

    public long? Stock { get; set; }

    public string? Image { get; set; }

    public long CategoryId { get; set; }

    public virtual ICollection<Cart> Carts { get; set; } = new List<Cart>();

    public virtual Category Category { get; set; } = null!;

    public virtual ICollection<FavoriteProduct> FavoriteProducts { get; set; } = new List<FavoriteProduct>();

    public virtual ICollection<Image> Images { get; set; } = new List<Image>();

    public virtual ICollection<Inventory> Inventories { get; set; } = new List<Inventory>();

    public virtual ICollection<AspNetUser> Users { get; set; } = new List<AspNetUser>();
}

File: Flowerapp/lib/model/product.dart
class Product {
  Product({
    required this.id,
    required this.values,
  });

  final String? id;
  final List<Value> values;

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json["\u0024id"],
      values: json["\u0024values"] == null
          ? []
          : (json["\u0024values"] is List
          ? List<Value>.from(json["\u0024values"].map((x) => Value.fromJson(x)))
          : [Value.fromJson(json["\u0024values"])]),
    );
  }

}

class Value {
  Value({
    required this.id,
    required this.idproduct,
    required this.name,
    required this.description,
    required this.price,
    required this.stock,
    required this.image,
    required this.categoryName,
  });

  final String? id;
  final int? idproduct;
  final String? name;
  final String? description;
  final int? price;
  final int? stock;
  final String? image;
  final String? categoryName;

  factory Value.fromJson(Map<String, dynamic> json){
    return Value(
      id: json["\u0024id"],
      idproduct: json["idproduct"],
      name: json["name"],
      description: json["description"],
      price: json["price"],
      stock: json["stock"],
      image: json["image"],
      categoryName: json["categoryName"],
    );
  }

}
